src/main/kotlin/com/exactpro/th2/codec/xml/XmlPipelineCodec.kt

```diff
@@ -77,18 +98,33 @@ open class XmlPipelineCodec(private val settings: XmlPipelineCodecSettings, xsdM
         }.build()
     }
 
+    private fun encodeOne(message: ParsedMessage): RawMessage {
+        val map = message.body
+        val xmlString = Xml.toXml(map)
+
+        validator.validate(xmlString.toByteArray())
```

Comment:
I think we can (and should) reuse this buffer when creating RawMessage below. Also, we must use correct charset when converting string to bytes